src/utils/options.py

```diff
@@ -0,0 +1,313 @@
+    exphan_to_radctrl_identity: str = "EXPHAN_RADCTRL_IDEN"
+    exphan_to_dsp_identity: str = "EXPHAN_DSP_IDEN"
+    dsp_to_radctrl_identity: str = "DSP_RADCTRL_IDEN"
+    dsp_to_driver_identity: str = "DSP_DRIVER_IDEN"
+    dsp_to_exphan_identity: str = "DSP_EXPHAN_IDEN"
+    dsp_to_dw_identity: str = "DSP_DW_IDEN"
+    dspbegin_to_brian_identity: str = "DSPBEGIN_BRIAN_IDEN"
+    dspend_to_brian_identity: str = "DSPEND_BRIAN_IDEN"
+    dw_to_dsp_identity: str = "DW_DSP_IDEN"
+    dw_to_radctrl_identity: str = "DW_RADCTRL_IDEN"
+    dw_to_rt_identity: str = "RT_DW_IDEN"
+    rt_to_dw_identity: str = "DW_RT_IDEN"
+    brian_to_radctrl_identity: str = "BRIAN_RADCTRL_IDEN"
+    brian_to_driver_identity: str = "BRIAN_DRIVER_IDEN"
+    brian_to_dspbegin_identity: str = "BRIAN_DSPBEGIN_IDEN"
+    brian_to_dspend_identity: str = "BRIAN_DSPEND_IDEN"
+
+    def parse_config(self):
+        # Read in config.ini file for current site
+        if not os.environ["BOREALISPATH"]:
+            raise ValueError("BOREALISPATH env variable not set")
+        if not os.environ['RADAR_ID']:
+            raise ValueError('RADAR_ID env variable not set')
+        path = f'{os.environ["BOREALISPATH"]}/config/' \
+               f'{os.environ["RADAR_ID"]}/' \
+               f'{os.environ["RADAR_ID"]}_config.ini'
+        try:
+            with open(path, 'r') as data:
+                raw_config = json.load(data)
+        except IOError:
+            print(f'IOError on config file at {path}')
+            raise
+
+        # Initialize all options from config file
+        self.site_id = raw_config['site_id']
+        self.main_antenna_count = int(raw_config["main_antenna_count"])
+        self.intf_antenna_count = int(raw_config["intf_antenna_count"])
+
+        # Parse N200 array and calculate which main and intf antennas operating
+        self.num_n200s = 0
+        self.n200_addrs = []
+        self.main_antennas = []
+        self.intf_antennas = []
+        for n200 in raw_config["n200s"]:
+            rx = bool(n200["rx"])
+            tx = bool(n200["tx"])
+            rx_int = bool(n200["rx_int"])
+            if rx or tx:
+                try:
+                    main_antenna_num = int(n200["main_antenna"])
+                except ValueError:
+                    errmsg = "main_antenna must be an integer if rx or tx is true"
+                    raise ValueError(errmsg)
+                self.main_antennas.append(main_antenna_num)
+            if rx_int:
+                try:
+                    intf_antenna_num = int(n200["intf_antenna"])
+                except ValueError:
+                    errmsg = "main_antenna must be an integer if rx or tx is true"
+                    raise ValueError(errmsg)
+                self.intf_antennas.append(intf_antenna_num)
+            if rx or tx or rx_int:
+                self.n200_addrs.append(n200["addr"])
+                self.num_n200s += 1
+        self.main_antennas.sort()
```

Comment:
Do we need to sort the IP addresses as well? I'm thinking that we need to preserve the mapping between n200 and antenna so that we can correctly beamform

Reply:
`n200_addrs` isn't used with any of the python files, I only added it here to check that there aren't any repeat IPs. The beamforming stuff must be done within the usrp driver since thats the only module that uses it